In [2]:
import pandas as pd

In [5]:
def add_unique_location_column(df, lat_col='latitude', lon_col='longitude'):
    """
    Adds a new column to the DataFrame with unique location names (e.g., 'Location 1', 'Location 2')
    based on latitude and longitude values.

    Args:
        df (pd.DataFrame): The DataFrame containing latitude and longitude columns.
        lat_col (str): Name of the column for latitude. Default is 'latitude'.
        lon_col (str): Name of the column for longitude. Default is 'longitude'.

    Returns:
        pd.DataFrame: The DataFrame with a new column 'location'.
    """
    # Create a unique identifier for each latitude-longitude pair
    unique_coords = {}
    location_counter = 1

    def get_location(lat, lon):
        nonlocal location_counter
        coord = (lat, lon)
        if coord not in unique_coords:
            unique_coords[coord] = f"Location {location_counter}"
            location_counter += 1
        return unique_coords[coord]

    df['location'] = df.apply(lambda row: get_location(row[lat_col], row[lon_col]), axis=1)
    return df

# Egypt

In [4]:
df = pd.read_csv('Data/irradiation_measurements/CBE_Data/CBE_Egypt.csv')
df['MEASURE_DATE'] = pd.to_datetime(df['MEASURE_DATE'])
df.set_index('MEASURE_DATE', inplace=True)
df.sort_index(inplace=True)
df.head()

,MEASURE_VALUE,SITE_COUNTRY,SITE_LATITUDE,SITE_LONGITUDE
MEASURE_DATE,,,,
2023-09-01,6428.052,Egypt,29.887561,32.460447
2023-09-02,6601.826,Egypt,29.887561,32.460447
2023-09-03,6600.638,Egypt,29.887561,32.460447
2023-09-04,6391.094,Egypt,29.887561,32.460447
2023-09-05,6373.021,Egypt,29.887561,32.460447


In [7]:
df_loc = add_unique_location_column(df, lat_col='SITE_LATITUDE', lon_col='SITE_LONGITUDE')
df_loc.head()

,MEASURE_VALUE,SITE_COUNTRY,SITE_LATITUDE,SITE_LONGITUDE,location
MEASURE_DATE,,,,,
2023-09-01,6428.052,Egypt,29.887561,32.460447,Location 1
2023-09-02,6601.826,Egypt,29.887561,32.460447,Location 1
2023-09-03,6600.638,Egypt,29.887561,32.460447,Location 1
2023-09-04,6391.094,Egypt,29.887561,32.460447,Location 1
2023-09-05,6373.021,Egypt,29.887561,32.460447,Location 1


In [10]:
df_loc['location'].unique()

array(['Location 1'], dtype=object)

In [11]:
print(f"Location 1: {df_loc[df_loc['location'] == 'Location 1'][['SITE_LATITUDE', 'SITE_LONGITUDE']].iloc[0].values}")

Location 1: [29.887561 32.460447]


In [12]:
print(f"First timestamp: {df_loc.index[0]}")
print(f"Last timestamp: {df_loc.index[-1]}")

First timestamp: 2023-09-01 00:00:00
Last timestamp: 2024-11-25 00:00:00


Egypt has one location with cordinates (29.887561, 32.460447) with data from 2023-09-01 to 2024-11-25

# Ghana

In [ ]:
df = pd.read_csv('Data/irradiation_measurements/CBE_Data/CBE_.csv')